libraries/apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/api/Service.kt

```diff
@@ -808,54 +807,76 @@ interface Service {
      * Connects the generated sources to the given Kotlin source set.
      * Throws if the Kotlin plugin is not applied
      *
-     * @param name: the name of the source set. For an example, "commonTest"
+     * @param name the name of the source set. For an example, "commonTest"
      */
     fun connectToKotlinSourceSet(name: String)
 
     /**
      * Connects the generated sources to the given Java source set.
      * Throws if the Java plugin is not applied
      *
-     * @param name: the name of the source set. For an example, "test"
+     * @param name the name of the source set. For an example, "test"
      */
     fun connectToJavaSourceSet(name: String)
 
+    /**
+     * Connects the generated sources to the given Android Component.
+     *
+     * @param component the component to connect to.
+     * - When using AGP8, component must be an instance of [com.android.build.gradle.api.BaseVariant]
+     * - When using AGP9+, component must be an instance of [com.android.build.api.variant.Component]
+     *
+     * Note [component]  is of type [Any] because [DirectoryConnection] can be used in non-Android projects,
+     * and we don't want the class to fail during loading because of a missing symbol in that case.
+     */
+    fun connectToAndroidComponent(component: Any, kotlin: Boolean)
+
+    /**
+     * Connects the generated sources to all the Variants.
+     */
+    fun connectToAndroidVariants(kotlin: Boolean)
+
+    /**
+     * Connects the generated sources to all the TestComponents.
+     */
+    fun connectToAndroidTestComponents(kotlin: Boolean)
+
+    /**
+     * Connects the generated sources to the given Android variant.
+     */
+    @Deprecated("use connectToAndroidComponent() instead", ReplaceWith("connectToAndroidComponent(variant)"))
```

Comment:
Nit: I guess the replacement won't work because it needs a value for `kotlin`. Maybe it should default to true?

Reply:
https://github.com/apollographql/apollo-kotlin/pull/6703/commits/f86dd33ce288bad9e5e43504ebd49126028af2b3